src/adapter/src/optimize/index.rs

```diff
@@ -149,24 +148,6 @@ impl Optimize<Index> for Optimizer {
         let mut df_builder = DataflowBuilder::new(state, self.compute_instance.clone());
         let mut df_desc = MirDataflowDescription::new(full_name.to_string());
 
-        // In EXPLAIN mode we should configure the dataflow builder to
-        // not consider existing indexes that are identical to the one that
-        // we are currently trying to explain.
```

Comment:
What was the purpose of this code? Was it from the days when we didn't have `EXPLAIN` for catalog objects, but only for `CREATE ...`?

Reply:
This is because when I wrote `EXPLAIN CREATE INDEX` I added a test in `tpch_create_index.slt` which ran

```sql
EXPLAIN WITH(humanized_exprs, arity, join_impls)
CREATE DEFAULT INDEX ON Q01;
```

**after**

```
CREATE DEFAULT INDEX ON Q01
```

and I thought that producing a plan that just reads from the already existing `q01_primary_idx` index didn't make much sense, so I wanted to exclude identical indexes in order to get a plan with the optimized `Q01` view.

```
materialize.public.q01_primary_idx:
  ArrangeBy keys=[[#0{l_returnflag}, #1{l_linestatus}]]
    ReadIndex on=q01 q01_primary_idx=[*** full scan ***]
```

I now realize that this is exactly what the optimizer will actually do (since the parser / planner don't reject identical `CREATE INDEX` statements), so it makes more sense the remove this code.

This is one more reason to implement dedicated `EXPLAIN REPLAN` syntax for exploring alternative optimization paths for existing materialized views / indexes with `bin/mzexplore`.